app/os_js.go

```diff
@@ -24,6 +24,15 @@ import (
 
 type ViewEvent struct{}
 
+type contextStatus int
+
+const (
+	contextStatusOkay contextStatus = 0
+	contextStatusLost contextStatus = 1 << (iota + 1)
+	contextStatusRestored
+	contextStatusDestroyed
+)
```

Comment:
Do you ever need more than one context status at the same time? I don't see anywhere. If not, convert to regular enum and avoid bit set operations.

Reply:
On `func (c *glContext) Refresh() error` it checks if it's destroyed and it's restored or not. So, it gives `gpu.ErrDeviceLost` and flags as `Destroyed`. Then, when restored, it will check if it's `Restored` and `Destroyed`, otherwise it will destroy first.

Reply:
Indeed, I had missed the bit manipulations happening in Refresh. See my other comment to why I still believe a state enum is enough.